olive/passes/onnx/graph_surgeries.py

```diff
@@ -78,50 +82,23 @@ def __init__(self, old_names: List[str], new_names: List[str]):
         self.old_names = old_names
         self.new_names = new_names
 
-    def __call__(self, model: ModelProto):
-        for old_name, new_name in zip(self.old_names, self.new_names):
-            for node in model.graph.node:
-                for idx, input_name in enumerate(node.input):
-                    if input_name == old_name:
-                        node.input[idx] = new_name
-
-                for idx, output_name in enumerate(node.output):
-                    if output_name == old_name:
-                        node.output[idx] = new_name
-
-            for idx, graph_input in enumerate(model.graph.input):
-                if graph_input.name == old_name:
-                    model.graph.input[idx].name = new_name
-
-            for idx, graph_output in enumerate(model.graph.output):
-                if graph_output.name == old_name:
-                    model.graph.output[idx].name = new_name
-        return model
+    def call_ir(self, model: ir.Model) -> ir.Model:
+        replacement = dict(zip(self.old_names, self.new_names))
+        for input in model.graph.inputs:
```

Comment:
Using ONNX IR is more efficient and simple: this now becomes an `O(#input)` operation.